### PR TITLE
gdal: better mingw support + sqlite3 by default + bump dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **gdal/all**

Several fixes are addressed in one PR, since GDAL is always hard to build in CI of CCI:

- fix build issues with MinGW:
  - `zlib` can be silently disabled because upstream `configure` use AC_CHECK_LIB  with -lz, but libz is not the name of zlib in conan recipe for MinGW.
  - surprisingly, `pkgconf` fails to find .pc files on Windows only, so `PKG_CONFIG_PATH` is forced with unix path of build folder (unix path is important, I tried without and it fails).
  - ~when we ask for shared GDAL, libtool fallbacks to an ill-formed static lib, because it wants all dependencies of GDAL to be shared. This expectation of libtool is so wrong, I don't know why it works like this.
  Anyway, generated libtool -after configure step- can be fixed to not perform these checks, and link static lib into the shared lib (we just need to change the value of `deplibs_check_method` variable to be `pass_all`).~ Maybe we could fix `libtool.m4` in `libtool` recipe instead (but GDAL has also its own copy of `libtool.m4`, so I don't know if it would be sufficient => EDIT: yes it works) => EDIT: fix is delegated to libtool recipe https://github.com/conan-io/conan-center-index/pull/5381
- inject environment variables in autotools build (for LD_LIBRARY_PATH & PATH): useful when build requirements executables could have be linked with shared libs, like `pkgconf` for example.
- add a verification of `libtiff` version, gdal needs libtiff >= 4.0.0, otherwise it fallbacks to vendored libtiff (and fails because we remove those files).
- enable `with_sqlite3` option by default, it's the default value in upstream `ax_lib_sqlite3.m4` actually.
- tiny fix for elastic driver compilation in 3.2.1 with gcc < 5 (upstream PR: https://github.com/OSGeo/gdal/pull/3440)
- bump dependencies (the most important being `proj`, major version is updated to latest).
- ~try to build gdal 3.1.2 with Visual Studio < 2019. At least I can tell that it works in github action for Visual Studio 2017: https://github.com/SpaceIm/conan-gdal/runs/2474046858?check_suite_focus=true.~ It should work if Visual Studio 2015 & 2017 agents have ATL installed now (needs https://github.com/conan-io/conan-center-index/issues/4195) => well not yet fixed in CI of CCI

Since this PR rebuild all versions, it should fix these PR:
closes https://github.com/conan-io/conan-center-index/issues/5017
closes https://github.com/conan-io/conan-center-index/issues/4902

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
```
c:\j\w\buildsinglereference@9\.conan\data\gdal\3.2.1\_\_\build\6dd24450bc2fe41d8b46bb790a030530df1417a0\source_subfolder\port\cpl_aws_win32.cpp(41): fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory
```

Well, ATL still not installed, I tried at least 😢

File: recipes/gdal/all/conanfile.py [hunk 1]
```diff
@@ -377,15 +380,29 @@ def _patch_sources(self):
...
+        if self.settings.compiler != "Visual Studio":
+            configure_ac = os.path.join(self._source_subfolder, "configure.ac")
+            # Workaround for nc-config not packaged in netcdf recipe (gdal relies on it to check nc4 and hdf4 support in netcdf):
+            if self.options.with_netcdf and self.options["netcdf"].netcdf4 and self.options["netcdf"].with_hdf5:
+                tools.replace_in_file(configure_ac,
+                                      "NETCDF_HAS_NC4=no",
+                                      "NETCDF_HAS_NC4=yes")
+            # Fix zlib checks and -lz injection to ensure to use external zlib and not fail others checks
+            if self.options.get_safe("with_zlib", True):
+                zlib_name = self.deps_cpp_info["zlib"].libs[0]
```
Comment (prince-chrismc) [inline recipes/gdal/all/conanfile.py:399]: Should we assert if theres more than one? For Some very weird consumer override

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/gdal/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/gdal/all/conanfile.py:399]: No, it doesn't matter.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**